contracts/src/Gateway.sol

```diff
@@ -265,11 +283,16 @@ contract Gateway is IGateway, IInitializable {
             revert InvalidAgentExecutionPayload();
         }
 
-        bytes memory call = abi.encodeCall(AgentExecutor.execute, params.payload);
+        (AgentExecuteCommand command, bytes memory commandParams) =
```

Comment:
Similar to my other recent comment, there isn't much sense in having an `AgentExecuteCommand` enum anymore. Instead, mintToken and TransferToken are just normal commands like `upgrade`, etc.

Reply:
https://github.com/Snowfork/snowbridge/pull/1155/commits/b2a666ac836fd926e087580ace88acb98e58a0d4